### PR TITLE
cookie -> sessionStorage changes

### PR DESCRIPTION
**Pull request by joekuttickal**
https://app.shortcut.com/radarlabs/story/10853/port-data-stored-in-cookies-to-sessionstorage-for-radar-js-sdk

### CODE REVIEW

**Comment by jkao** [comment]
how did you QA this?

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,19 +2,19 @@ version: 2
...
       - run:
           name: Update npm
           command: 'npm install -g npm@latest'
       - restore_cache:
-          key: dependency-cache-{{ checksum "package.json" }}
+          key: dependency-cache-{{ checksum "package.json" }}-v2
       - run:
           name: Install Dependencies
-          command: npm install
+          command: npm ci --legacy-peer-deps
```
[the remark this answers is not in the document]
Comment (joekuttickal) [reply .circleci/config.yml:15]: i was running into peer dependency errors when building it on ci: 
Screen Shot 2022-07-06 at 4 02 42 PM

**Comment by jkao** [comment]
this UNAUTHORIZED issue seems to imply it isn't QA'ed.

looks like the header `-H 'Authorization: true'` is incorrect. going to investigate.

**Comment by jkao** [comment]
I have a fix for the above; there was logic error in the sessionStorage helper.

image

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,17 +2,14 @@ version: 2
...
       - checkout
-      - run:
-          name: Update npm
-          command: 'npm install -g npm@latest'
       - restore_cache:
           key: dependency-cache-{{ checksum "package.json" }}
       - run:
           name: Install Dependencies
-          command: npm install
+          command: npm ci --force
```
Comment (corypisano) [inline .circleci/config.yml:12]: why force?

**Review by corypisano — DISMISSED** [review]
Looks good, just want to check about the package-lock changes (9800 additions)

We can let this bake on staging a bit, we'll need a couple frontend changes for simulator since it explicitly sets some cookies for the sdk

File: .circleci/config.yml (hunk 2 above)
Comment (jkao) [reply .circleci/config.yml:12]: so `npm ci` revealed an issue in our dependency tree:

```
    "rollup": "^2.3.3",
    "rollup-plugin-babel": "^4.4.0",
    "rollup-plugin-commonjs": "^10.1.0",
    "rollup-plugin-node-resolve": "^5.2.0",
    "rollup-plugin-uglify": "^6.0.4",
```
    
`rollup-plugin-uglify` on the _latest_ version (6.0.4) expects a version of `rollup` < 2. we are using 2.3.3. the upgrade was an attempt to see if they had updated the rules.

given that these are dev dependencies and the commands appear to working, I added the `--force` flag to ignore that check.

the other option is to downgrade rollup, which is not great, either.

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,9432 @@
 {
   "name": "radar-sdk-js",
-  "version": "3.2.1",
-  "lockfileVersion": 1,
+  "version": "3.3.0",
+  "lockfileVersion": 2,
```
[the remark this answers is not in the document]
Comment (jkao) [reply package-lock.json:4]: yes, I used `npm` on node 16.5.0 which should match our build configs, and this is the result.

**Comment by jkao** [comment]
I'm going to update this to a `beta` version "just in case"

**Pull request merged**